src/Common/getNumberOfPhysicalCPUCores.cpp

```diff
@@ -36,20 +42,73 @@ static uint32_t getCGroupLimitedCPUCores(unsigned default_cpu_count)
 }
 #endif
 
-static unsigned getNumberOfPhysicalCPUCoresImpl()
+/// Returns number of physical cores, unlike std::thread::hardware_concurrency() which returns the logical core count. With 2-way SMT
+/// (HyperThreading) enabled, physical_concurrency() returns half of of std::thread::hardware_concurrency(), otherwise return the same.
+#if defined(__x86_64__) && defined(OS_LINUX)
+unsigned physical_concurrency()
+try
```

Comment:
Function-try-block looks strange to me w/o a constructor, but I guess it's just stylistic nuances.

Reply:
It is a stylistic nuance :)